flow/flow.go

```diff
@@ -372,6 +380,35 @@ func (f *Flow) Update(now int64, packet *gopacket.Packet, length int64) {
 	if f.TCPFlowMetric != nil {
 		f.updateTCPMetrics(packet)
 	}
+	f.updateRTT(packet)
+}
+
+func (f *Flow) updateRTT(packet *gopacket.Packet) {
+	if f.Link != nil && f.RTT == 0 {
+		ethernetLayer := (*packet).Layer(layers.LayerTypeEthernet)
+		ethernetPacket, ok := ethernetLayer.(*layers.Ethernet)
+		if ok && (f.Link.A == ethernetPacket.DstMAC.String()) {
+			f.RTT = (*packet).Metadata().Timestamp.UnixNano() - f.state.link1stPacket
```

Comment:
I would prefer to move this in the updateMetricsWithLinkLayer and updateMetricsWithNetworkLayer, it will avoid the same cast switch/case

Reply:
done